### PR TITLE
[FAB-16895] Move ledger benchmark tests from fabric-test repo

### PR DESCRIPTION
**Pull request by manish-sethi**
Signed-off-by: manish <manish.sethi@gmail.com>

#### Type of change

- Improvement (improvement to code, performance, etc)
- Test update

#### Description

Move ledger benchmark tests from _fabric-test_ repo to _fabric_ repo under _core/ledger_ package
so that it can eventually be deleted from _fabric-test_ repo.

### CODE REVIEW

File: core/ledger/kvledger/benchmark/chainmgmt/sanity_test.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package chainmgmt
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+// TestChainMgmt is a basic sanity check test to catch any errors that could be caused by changes in the ledgermgmt ot kvledger packages
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/sanity_test.go]:
```suggestion
// TestChainMgmt is a basic sanity check test to catch any errors that could be caused by changes in the ledgermgmt or kvledger packages
```

File: core/ledger/kvledger/benchmark/chainmgmt/chains.go [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+	ledgermgmtInitializer.Config.HistoryDBConfig.Enabled = true
+	if os.Getenv("useCouchDB") == "true" {
+		ledgermgmtInitializer.Config.StateDBConfig.StateDatabase = "CouchDB"
+		ledgermgmtInitializer.Config.StateDBConfig.CouchDB = &couchdb.Config{
+			Address:            "localhost:5984",
+			RedoLogPath:        filepath.Join(dataDir, "couchdbRedologs"),
+			UserCacheSizeMBs:   500,
+			MaxBatchUpdateSize: 500,
+		}
+	}
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/chains.go]: Seems that we should be using the same patterns that we use in the rest of the tests and avoid hard-coded hosts and ports of cloud and rely on what's in COUCHDB_ADDR.

File: core/ledger/kvledger/benchmark/chainmgmt/chains.go [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+	return chains
+}
+
+func (m *chainsMgr) waitForChainsToExhaustAllBlocks() {
+	m.wg.Wait()
+	m.ledgerMgr.Close()
+}
+
+// Chain extends ledger.PeerLedger and the experiments invoke ledger functions via a chain type
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/chains.go]:
```suggestion
// Chain embeds ledger.PeerLedger and the experiments invoke ledger functions via a chain type
```
There is no inheritance in `go`.

File: core/ledger/kvledger/benchmark/chainmgmt/conf.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package chainmgmt
+
+// ChainMgrConf captures the configurations meant at the level of chainMgr
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/conf.go]:
```
// ChainMgrConf captures the configurations meant at the level of chainMgr
```
Not sure I grok that.

File: core/ledger/kvledger/benchmark/chainmgmt/conf.go [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+// DataDir field specifies the filesystem location where the chains data is maintained
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/chainmgmt/conf.go]:
```
// DataDir field specifies the filesystem location where the chains data is maintained
// NumChains field specifies the number of chains to instantiate
```
If this are field doc, shouldn't they be with the fields? (Here and below.)

File: core/ledger/kvledger/benchmark/experiments/util.go [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+	"john",
+}
+
+//TestValue is a struct for holding the test value
+type TestValue struct {
+	Value string
+}
+
+func constructKey(keyNumber int) string {
+	return fmt.Sprintf("%s%09d", "key_", keyNumber)
```
Comment (sykesm) [inline core/ledger/kvledger/benchmark/experiments/util.go]:
```suggestion
	return fmt.Sprintf("key_%09d", keyNumber)
```

**Review by sykesm — COMMENTED** [review]
Mostly nits with two exceptions:

1. Please use an actual temporary directory as returned by ioutil.TempDir or similar.
2. Please use COUCHDB_ADDR instead of hard-coded address

Biggest concern overall is the amount of bash that's being used to glue this together; it's kind of insane for what it's doing. Even things like using the output of the test command to look for the `FAIL` string instead of honoring the exit code.

**Review by denyeart — APPROVED** [review]

**Pull request merged**